### PR TITLE
Change log level at runtime

### LINKED ISSUE #29: Set the log level at runtime
# Description
Currently the log level is [hardcoded](https://github.com/HaaLeo/swarmlib/blob/4a21dd2e8ab0d8f0928be358ba7ee360735c646d/swarmlib/__main__.py#L22) to be `INFO`. Allowing it to be set by the users during runtime could be beneficial for future bug reports. 

# Proposal
* Either add a top level flag `--log-level` which can be set to one of [python's log level](https://docs.python.org/3/howto/logging.html)
* Or add a (boolean) flag `--verbose` similar how `sklearn` uses it. ([reference](https://sklearn.org/modules/generated/sklearn.linear_model.Lars.html#sklearn.linear_model.Lars))

### PR DESCRIPTION
**Pull request by nkoutsov**
# Summary

Allow changing logging level at runtime by adding a command line argument.
# Changes

* Add --log-level optional argument
* If --log-level set, change level to given value

# Related Issues

Fixes #29

### CODE REVIEW

File: swarmlib/__main__.py [hunk 1]
```diff
@@ -58,6 +58,10 @@ def run_swarm():
         default=None,
         type=int,
         help='Used to set the initial state of the random bit generator (default None)')
+    parser.add_argument(
+        '--log-level',
+        default="INFO",
```
Comment (HaaLeo) [inline swarmlib/__main__.py]: Please use `'` instead of `"` to make pylint pass.

File: swarmlib/__main__.py [hunk 2]
```diff
@@ -72,6 +76,10 @@ def run_swarm():
     args = vars(parser.parse_args())
     if args:
         algorithm = args.pop('func')
+        loglevel = args.pop('log_level', "INFO")
```
Comment (HaaLeo) [inline swarmlib/__main__.py]: Please use `'` instead of `"` to make pylint pass.

File: swarmlib/__main__.py (hunk 1 above)
Comment (HaaLeo) [inline swarmlib/__main__.py]: Can you make this a [choice](https://docs.python.org/3/library/argparse.html#choices) of `['debug', 'info', 'warning', 'error', 'critical']` ? Then the parser automatically deals with the error handling.

File: swarmlib/__main__.py [hunk 3]
```diff
@@ -72,6 +76,10 @@ def run_swarm():
...
+        level = getattr(logging, loglevel.upper(), None)
+        if level:
```
Comment (HaaLeo) [inline swarmlib/__main__.py]: This check should be unnecessary when the argument is a _choice_ with a default value

File: swarmlib/__main__.py [hunk 4]
```diff
@@ -72,6 +76,10 @@ def run_swarm():
     args = vars(parser.parse_args())
     if args:
         algorithm = args.pop('func')
+        loglevel = args.pop('log_level', "INFO")
+        level = getattr(logging, loglevel.upper(), None)
```
Comment (HaaLeo) [inline swarmlib/__main__.py]: The default value here should also be unnecessary when we use a _choice_ in the parser

**Review by HaaLeo — CHANGES_REQUESTED** [review]
Hi @nkoutsov thank you for your contribution. Bellow there are just a few small requests that restrict the usage a little bit more.

**Comment by nkoutsov** [comment]
> Hi @nkoutsov thank you for your contribution. Bellow there are just a few small requests that restrict the usage a little bit more.

Hello, @HaaLeo . I just made some changes. Take a look and let me know

File: swarmlib/__main__.py [hunk 5]
```diff
@@ -58,6 +58,11 @@ def run_swarm():
         default=None,
         type=int,
         help='Used to set the initial state of the random bit generator (default None)')
+    parser.add_argument(
+        '--log-level',
+        default='info',
+        choices=['debug', 'info', 'warning', 'error', 'critical'],
+        help='Set the log level')
```
Comment (HaaLeo) [inline swarmlib/__main__.py]: Just one last single bit: can you add the default in the help description, too? Similar to like it is done with the `--seed` option above. Then we are good to go

**Review by HaaLeo — CHANGES_REQUESTED** [review]

File: swarmlib/__main__.py (hunk 5 above)
Comment (nkoutsov) [reply swarmlib/__main__.py]: Done :)

**Review by HaaLeo — APPROVED** [review]
